src/core/vector/qgsvectorlayerfeatureiterator.cpp

```diff
@@ -1341,6 +1341,12 @@ void QgsVectorLayerFeatureIterator::createExpressionContext()
   mExpressionContext->appendScope( QgsExpressionContextUtils::projectScope( QgsProject::instance() ) ); // skip-keyword-check
   mExpressionContext->appendScope( new QgsExpressionContextScope( mSource->mLayerScope ) );
   mExpressionContext->setFeedback( mRequest.feedback() );
+  // We need to keep the map settings scope if set in the original context
+  const int settingsScopeIndex = mRequest.expressionContext()->indexOfScope( QStringLiteral( "Map Settings" ) );
```

Comment:
I don't understand the reason for this change -- could you break this pr up into separate atomic commits so that I can understand better?